dsgrid/utils/spark.py

```diff
@@ -351,19 +357,19 @@ def write_dataframe_and_auto_partition(df, filename, partition_size_mb=128, colu
     total_size = sum((x.stat().st_size for x in filename.glob("*.parquet")))
     desired = math.ceil(total_size / partition_size_bytes)
     actual = df.rdd.getNumPartitions()
-    if actual > desired:
+    if abs(actual - desired) / min(actual, desired) * 100 < rtol_pct:
```

Comment:
A bit confused by the min() func here, because it anchors around the smaller value. This is the same as saying, do nothing if partitions falls within 50%-150% of the smaller of actual vs. desired.

Reply:
I chose this because I think we care more about the relative difference than absolute. If anything, perhaps we should anchor on the larger value because it will be more lenient. But I'm open if there are standard ways of expressing this.

Reply:
I agree that you are correct and fixed it.